modules/reindex/src/internalClusterTest/java/org/opensearch/index/codec/MultiCodecReindexIT.java

```diff
@@ -61,72 +57,94 @@ public void testReindexingMultipleCodecs() throws InterruptedException, Executio
             "lz4",
             "BEST_SPEED"
         );
-
-        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
-            assertReindexingWithMultipleCodecs(codec.getKey(), codec.getValue(), codecMap);
-        }
-
-    }
-
-    private void assertReindexingWithMultipleCodecs(String destCodec, String destCodecMode, Map<String, String> codecMap)
-        throws ExecutionException, InterruptedException {
-
-        final String index = "test-index" + destCodec;
-        final String destIndex = "dest-index" + destCodec;
-
-        // creating source index
-        createIndex(
-            index,
-            Settings.builder()
-                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
-                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
-                .put("index.codec", "default")
-                .put("index.merge.policy.max_merged_segment", "1b")
-                .build()
```

Comment:
Instead of creating a separate `index` for all codec tests, we've replaced it with a unified "source" index containing docs encoded with all kinds of codecs.

Reply:
https://github.com/opensearch-project/OpenSearch/pull/10980/files#r1375666873